src/libraries/System.Private.CoreLib/src/System/Text/StringBuilder.cs

```diff
@@ -1205,28 +1170,48 @@ public StringBuilder Append(char[]? value)
         {
             if (value?.Length > 0)
```

Comment:
Can we get rid of the null and length checks, too?  `value.AsSpan()` will already return an empty span for null and it'll short-circuit for empty.  I doubt we should be optimizing for the null/empty case.